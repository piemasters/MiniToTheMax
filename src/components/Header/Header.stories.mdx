import {
  Story,
  Preview,
  Props,
  Source,
  Description,
  Meta,
} from '@storybook/addon-docs/blocks';
import { Header } from './Header';

<Meta
  title="Components/Header"
  component={Header}
  parameters={{ jest: ['header'] }}
  argTypes={{
    title: {
      name: 'title',
      description: 'Website title',
      defaultValue: 'MiniToTheMax',
      control: { type: 'text' },
    },
    logo: {
      name: 'logo',
      description: 'Path to the logo image',
      defaultValue: '',
      control: { type: 'text' },
    },
    pages: {
      name: 'pages',
      description: 'The hyperlinks to display',
      defaultValue: [],
      control: { type: 'array' },
    },
  }}
/>

# Header

With `MDX` we can define a story for `Header` right in the middle of our
markdown documentation.

<Preview>
  <Story
    name="default"
    args={{
      title: 'MiniToTheMax',
      logo: './images/logo.png',
      pages: [
        { name: 'Home', url: '/' },
        { name: 'Blog', url: '/blog' },
        { name: 'Showcase', url: '/showcase' },
        { name: 'Backlog', url: '/backlog' },
      ],
    }}
  >
    {(args) => <Header {...args} />}
  </Story>
</Preview>

# Props

<Props story="default" />
