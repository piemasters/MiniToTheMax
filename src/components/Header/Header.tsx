import React, { FC } from 'react';

import { PageLink } from '../PageLink';
import type { NavLink } from '../../types';

export interface HeaderProps {
  title: string;
  logo: string;
  pages: NavLink[];
}

export const Header: FC<HeaderProps> = ({ title, logo, pages }) => {
  return (
    <div className="bg-gray-100 px-4 py-8 mb-8" data-testid="nav-header">
      <nav className="flex flex-col mx-auto max-w-3xl">
        <PageLink
          type={'paintDrip'}
          to={'/'}
          className="flex items-center text-gray-900 pb-4 no-underline"
        >
          <img src={logo} alt="Logo" width={60 + 'px'} />
          <h1 className="ml-4 mb-0">{title}</h1>
        </PageLink>
        <div className="flex list-none m-0">
          {pages.map((link: NavLink) => {
            return (
              <PageLink
                to={link.url}
                className="text-gray-500 text-sm font-bold mr-5 no-underline hover:text-red-500"
                key={link.name}
                linkActiveStyle="!text-red-500"
                type={'cover'}
              >
                {link.name}
              </PageLink>
            );
          })}
        </div>
      </nav>
    </div>
  );
};

export default Header;
