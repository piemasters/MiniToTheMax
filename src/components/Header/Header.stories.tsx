import type { Meta, StoryObj } from '@storybook/react-vite';
import { Header } from './Header';

const meta: Meta<typeof Header> = {
  component: Header,
  argTypes: {
    title: {
      name: 'title',
      description: 'Website title',
      defaultValue: 'MiniToTheMax',
      control: { type: 'text' },
    },
    logo: {
      name: 'logo',
      description: 'Path to the logo image',
      defaultValue: '',
      control: { type: 'text' },
    },
    pages: {
      name: 'pages',
      description: 'The hyperlinks to display',
      defaultValue: [],
      control: { type: 'array' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Header>;

export const Primary: Story = {
  args: {
    title: 'MiniToTheMax',
    logo: '/images/logo.webp',
    pages: [
      { name: 'Home', url: '/' },
      { name: 'Blog', url: '/blog' },
      { name: 'Showcase', url: '/showcase' },
      { name: 'Backlog', url: '/backlog' },
    ],
  },
};
