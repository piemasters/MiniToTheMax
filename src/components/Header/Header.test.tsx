import { getByTestId, render } from '@testing-library/react';

import { Header } from './Header';

describe('Header', () => {
  test('renders correctly', () => {
    const data = {
      site: {
        siteMetadata: {
          title: 'MiniToTheMax',
        },
      },
      file: {
        publicURL: '/static/logo-37f736b45fa7360612c71ff66ce83af8.png',
      },
    };

    const pages = [
      { name: 'Home', url: '/' },
      { name: 'Blog', url: '/blog' },
      { name: 'Showcase', url: '/showcase' },
      { name: 'Backlog', url: '/backlog' },
    ];

    const { container } = render(
      <Header
        title={data.site.siteMetadata.title}
        logo={data.file.publicURL}
        pages={pages}
      />
    );

    expect(
      getByTestId(container as HTMLElement, 'nav-header')
    ).toHaveTextContent('MiniToTheMax');

    expect(container).toMatchSnapshot();
  });
});
