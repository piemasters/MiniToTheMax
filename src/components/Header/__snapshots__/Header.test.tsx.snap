// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    class="bg-gray-100 px-4 py-8 mb-8"
    data-testid="nav-header"
  >
    <nav
      class="flex flex-col mx-auto max-w-3xl"
    >
      <a
        aria-current="page"
        class="flex items-center text-gray-900 pb-4 no-underline"
        data-testid="page-link"
        href="/"
      >
        <img
          alt="Logo"
          src="/static/logo-37f736b45fa7360612c71ff66ce83af8.png"
          width="60px"
        />
        <h1
          class="ml-4 mb-0"
        >
          MiniToTheMax
        </h1>
      </a>
      <div
        class="flex list-none m-0"
      >
        <a
          aria-current="page"
          class="text-gray-500 text-sm font-bold mr-5 no-underline hover:text-red-500 !text-red-500"
          data-testid="page-link"
          href="/"
        >
          Home
        </a>
        <a
          class="text-gray-500 text-sm font-bold mr-5 no-underline hover:text-red-500"
          data-testid="page-link"
          href="/blog"
        >
          Blog
        </a>
        <a
          class="text-gray-500 text-sm font-bold mr-5 no-underline hover:text-red-500"
          data-testid="page-link"
          href="/showcase"
        >
          Showcase
        </a>
        <a
          class="text-gray-500 text-sm font-bold mr-5 no-underline hover:text-red-500"
          data-testid="page-link"
          href="/backlog"
        >
          Backlog
        </a>
      </div>
    </nav>
  </div>
</div>
`;
